pkg/sbom/spdx/marshal_test.go

```diff
@@ -716,6 +758,17 @@ func TestMarshaler_Marshal(t *testing.T) {
 			spdxDoc, err := marshaler.Marshal(tc.inputReport)
 			require.NoError(t, err)
 
+			sort.Slice(spdxDoc.Packages, func(i, j int) bool {
+				return spdxDoc.Packages[i].PackageSPDXIdentifier < spdxDoc.Packages[j].PackageSPDXIdentifier
+			})
+			sort.Slice(tc.wantSBOM.Packages, func(i, j int) bool {
+				return tc.wantSBOM.Packages[i].PackageSPDXIdentifier < tc.wantSBOM.Packages[j].PackageSPDXIdentifier
+			})
```

Comment:
I think they are no longer needed after changing packages to slice.

Reply:
since the map is used to remove duplications I still needed it.

Reply:
Yes, it is relevant to this discussion.
https://github.com/aquasecurity/trivy/pull/4058#discussion_r1168709377